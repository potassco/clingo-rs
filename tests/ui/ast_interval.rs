// ast_interval
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let interval = ast::interval(&loc, term1, term2);
    drop(term1);
    drop(term2);
    let _end = interval;
}
