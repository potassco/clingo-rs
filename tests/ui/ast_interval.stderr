error[E0382]: use of moved value: `term1`
  --> tests/ui/ast_interval.rs:10:10
   |
7  |     let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
   |         ----- move occurs because `term1` has type `Term<'_>`, which does not implement the `Copy` trait
8  |     let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
9  |     let interval = ast::interval(&loc, term1, term2);
   |                                        ----- value moved here
10 |     drop(term1);
   |          ^^^^^ value used here after move
   |
help: consider cloning the value if the performance cost is acceptable
   |
9  |     let interval = ast::interval(&loc, term1.clone(), term2);
   |                                             ++++++++

error[E0382]: use of moved value: `term2`
  --> tests/ui/ast_interval.rs:11:10
   |
8  |     let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
   |         ----- move occurs because `term2` has type `Term<'_>`, which does not implement the `Copy` trait
9  |     let interval = ast::interval(&loc, term1, term2);
   |                                               ----- value moved here
10 |     drop(term1);
11 |     drop(term2);
   |          ^^^^^ value used here after move
   |
help: consider cloning the value if the performance cost is acceptable
   |
9  |     let interval = ast::interval(&loc, term1, term2.clone());
   |                                                    ++++++++
